programs/survival/graves.sc

```diff
@@ -0,0 +1,349 @@
+            return();
+        );
+
+
+        deathCrate= create_screen(player,'generic_9x5',str('%s\'s Death Crate',gdata:'name'),_(screen,player,action,data,outer(gdata),outer(block)) ->(
+
+            if(action=='pickup' || action== 'swap' || action=='quick_move' || action== 'throw' || action== 'pickup_all',
+                _delete_data_slot(gdata,data:'slot');
+                datafound=false;
+
+                c_for(i=0,i<41,i+=1,if(gdata:str(i)!=null, datafound=true));
+
+                if(datafound==false,
+                    _delete_data(pos(block));
+                    run(str('setblock %d %d %d minecraft:air',pos(block):0, pos(block):1, pos(block):2)); //for some reason /set doesnt seem to work. I might be doing bonk stuff somewhere
+                );
+            );
+        ));
+
+
+        if(screen_property(deathCrate,'open'),
+            c_for(i=0,i<41,i+=1,
+                item=gdata:str(i);
+                inventory_set(deathCrate,i,item:1,item:0,item:2);
+            );       
+        );
+        if(!gdata:'opened',modify(player,'add_xp',global_gdata:str(pos(block)):'xp'));
+        gdata:'opened'=true;
+    );
+);
+
+_owner_clicked(player,block) -> (
+
+    gdata=global_gdata:str(pos(block));
+    itemCount=0;
+    c_for(i=0,i<inventory_size(player),i+=1,
+        item=global_gdata:str(pos(block)):str(i);
+        if(item!=null && inventory_get(player,i)==null,
+            inventory_set(player,i,item:1,item:0,item:2);
+            _delete_data_slot(gdata,str(i)),
+         // when something is in slot
+            item!=null && inventory_get(player,i)!=null,
+            itemCount+=1;
+        );
+    );
+
+    if(itemCount == 0,
+        modify(player,'xp_level',gdata:'xp_level');
+        modify(player,'xp_progress',gdata:'xp_progress');
+        set(pos(block),'air');
+        _delete_data(pos(block));
+        return();          
+    );
+
+    deathCrate= create_screen(player,'generic_9x5',str('%s\'s Death Crate',gdata:'name'),_(screen,player,action,data,outer(gdata),outer(block)) ->(
+
+            if(action=='pickup' || action== 'swap' || action=='quick_move' || action== 'throw' || action== 'pickup_all',
+                _delete_data_slot(gdata,data:'slot');
+                datafound=false;
+            c_for(i=0,i<41,i+=1,if(gdata:str(i)!=null, datafound=true));
+
+            if(datafound==false,
+            _delete_data(pos(block));
+            modify(player,'xp_level',gdata:'xp_level');
+            modify(player,'xp_progress',gdata:'xp_progress');
+            run(str('setblock %d %d %d minecraft:air',pos(block):0, pos(block):1, pos(block):2));
+            );
+            );
+
+        ));
+
+    if(screen_property(deathCrate,'open'),
+
+            c_for(i=0,i<41,i+=1,
+                item=gdata:str(i);
+                inventory_set(deathCrate,i,item:1,item:0,item:2);
+            );      
+        );
+);
+
+
+
+
+//Utility Functions
+
+_request(global_gdata,block) -> (
```

Comment:
if this is a global variable, then why do you need to pass it as an argument? Surely you can just grab it from within the function? (unless there is a specific reason for passing it in as an argument ofc)

Reply:
This is plain old embarrassing xD, there is no reason. I overlooked this while going through the thing, also I dont even think its necessary to put this one-liner into a separate function. Imma change this too